transform-plugins/src/test/java/co/cask/hydrator/plugin/DecompressorTest.java

```diff
@@ -0,0 +1,114 @@
+
+  @Test
+  public void testSnappyCompress() throws Exception {
+    String decompressTester = "This is a test for testing snappy compression";
+    Transform<StructuredRecord, StructuredRecord> transform =
+      new Decompressor(new Decompressor.Config("a:SNAPPY", OUTPUT.toString()));
+    transform.initialize(null);
+
+    MockEmitter<StructuredRecord> emitter = new MockEmitter<>();
+    byte[] compressed = Snappy.compress(decompressTester.getBytes());
+    transform.transform(StructuredRecord.builder(INPUT)
+                          .set("a", compressed)
+                          .set("b", "2")
+                          .set("c", "3")
+                          .set("d", "4")
+                          .set("e", "5").build(), emitter);
+    String actual = emitter.getEmitted().get(0).get("a");
+    Assert.assertEquals(2, emitter.getEmitted().get(0).getSchema().getFields().size());
+    Assert.assertEquals(decompressTester, actual);
+  }
+
+  @Test
+  public void testZipCompress() throws Exception {
+    String decompressTester = "This is a test for testing zip compression";
+    Transform<StructuredRecord, StructuredRecord> transform =
+      new Decompressor(new Decompressor.Config("a:ZIP", OUTPUT.toString()));
+    transform.initialize(null);
+
+    MockEmitter<StructuredRecord> emitter = new MockEmitter<>();
+    byte[] compressed = zip(decompressTester.getBytes());
+    transform.transform(StructuredRecord.builder(INPUT)
+                          .set("a", compressed)
+                          .set("b", "2")
+                          .set("c", "3")
+                          .set("d", "4")
+                          .set("e", "5").build(), emitter);
+    String actual = emitter.getEmitted().get(0).get("a");
+    Assert.assertEquals(2, emitter.getEmitted().get(0).getSchema().getFields().size());
+    Assert.assertEquals(decompressTester, actual);
+  }
+
+  @Test
+  public void testGZipCompress() throws Exception {
+    String decompressTester = "This is a test for testing gzip compression";
+    Transform<StructuredRecord, StructuredRecord> transform =
+      new Decompressor(new Decompressor.Config("a:GZIP", OUTPUT.toString()));
+    transform.initialize(null);
+
+    MockEmitter<StructuredRecord> emitter = new MockEmitter<>();
+    byte[] compressed = gzip(decompressTester.getBytes());
+    transform.transform(StructuredRecord.builder(INPUT)
+                          .set("a", compressed)
+                          .set("b", "2")
+                          .set("c", "3")
+                          .set("d", "4")
+                          .set("e", "5").build(), emitter);
+    String actual = emitter.getEmitted().get(0).get("a");
+    Assert.assertEquals(2, emitter.getEmitted().get(0).getSchema().getFields().size());
+    Assert.assertEquals(decompressTester, actual);
+  }
+
+
+  private static byte[] gzip(byte[] input) throws IOException {
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
```

Comment:
Maybe only for GZIPOutputStream so you don;t have to call gzip.close explicitly.